client/platform/desktop/api/nativeServices.ts

```diff
@@ -0,0 +1,152 @@
+/**
+ * Native services that directly interact with the OS.
+ * I'm not sure whether I want to call these from render or main thread.
+ * IPC module is not ideal for transferring such large amounts of data.
+ * It would be reasonable to use HTTP either with the embedded server
+ * or perhaps a custom `local-file` protocol
+ *
+ * https://nklayman.github.io/vue-cli-plugin-electron-builder/guide/configuration.html#changing-the-file-loading-protocol
+ */
+
+import npath from 'path';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import fs from 'fs-extra';
+import moment from 'moment';
+import { TrackData } from 'vue-media-annotator/track';
+import { SaveDetectionsArgs, DatasetType } from 'viame-web-common/apispec';
+
+// Match examples:
+// result_09-14-2020_14:49:05.json
+// result.json
+const JsonFileName = /^result(_\d\d-\d\d-\d\d\d\d_\d\d:\d\d:\d\d)?\.json$/;
+const CsvFileName = /^viame-annotations.csv$/;
+const AuxFolderName = 'auxiliary';
+
+/**
+ * @param path a known, existing path
+ */
+async function loadJsonAnnotations(path: string): Promise<Record<string, TrackData>> {
+  const rawBuffer = await fs.readFile(path, 'utf-8');
+  const annotationData = JSON.parse(rawBuffer);
+  // TODO: validate json schema
+  return annotationData as Record<string, TrackData>;
+}
+
+/**
+ * @param path a known, existing path
+ */
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+async function loadCSVAnnotations(path: string): Promise<Record<string, TrackData>> {
+  // TODO: maybe come up with a way to invoke python csv -> json transcoding
+  // Otherwise we'll have to reimplement in Node.js
+  return {};
+}
+
+async function getDatasetBase(datasetId: string): Promise<{
+  datasetType: DatasetType;
+  basePath: string;
+  jsonFile: string | null;
+  directoryContents: string[];
+}> {
+  let datasetType: DatasetType = 'image-sequence';
+  const exists = fs.existsSync(datasetId);
+  if (!exists) {
+    throw new Error(`No dataset exists with path ${datasetId}`);
+  }
+  const stat = await fs.stat(datasetId);
+
+  if (stat.isDirectory()) {
+    datasetType = 'image-sequence';
+  } else if (stat.isFile()) {
+    datasetType = 'video';
+  } else {
+    throw new Error('Symlinks not supported');
+  }
```

Comment:
Symlink was just the first exception I could think of.

https://stackoverflow.com/questions/15630770/node-js-check-if-path-is-file-or-directory

There are also FIFOs, block devices, and all sorts of other stuff.  I should change the error message.

Reply:
I can't find difinitive docs on whether isFile and isDirectory are mutually exclusive 100% coverage, or if there are other cases, but I'd rather be unsure and have the error.

Reply:
Fixed b35bea9

Reply:
I don't think I was clear in my comment.  The main issue is that it thinks everything is a image-sequence because all datasets are folders now.  So when I was logging this the datasetType here would be an image-sequence but then in main.ts it would be a video.  My main thing was that this whole section might not be necessary because in desktop you don't support single files.  I might be wrong in my interpretation of that.

Reply:
Oh!  Duh.   Sorry I didn't pick up on this.

I'll get that fixed.  Thank you.